ingestor/service.go

```diff
@@ -381,6 +391,18 @@ func (s *Service) HandleTransfer(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusAccepted)
 }
 
+func (s *Service) Shutdown() error {
+	if err := s.metrics.Close(); err != nil {
+		return err
+	}
+
+	if err := s.UploadSegments(); err != nil {
```

Comment:
I just noticed that this func has a hard coded 30s timeout to upload.  I'd suggest refactoring that to take a `Context` and then you can pass int a `context.WithTimeOut(5 * time.Minutes)` to give it more time to complete uploads.  The 30s timeout is a remnant from the SIGTERM handler that we don't use anymore.

Reply:
sounds good. I've added context.Timeout in the uploadWrites method.